### ISSUE TITLE
Cleanup signals emitted when navigating to anchor inside of already loaded document

### ISSUE THREAD

**Issue by Emdek** [annulen/webkit#659]
Currently navigating to such anchor causes `QWebFrame::loadFinished()` to be emitted twice but no signals from page and no `QWebFrame::loadStarted()`.
Moreover, there is call to `QWebPage::acceptNavigationRequest()` with type `QWebPage::NavigationTypeOther` (A navigation to another document using a method not listed above.) which is misleading (it's the same document).
Meanwhile QtWebEngine does nothing of the above and simply scrolls document to anchor.

**Comment by annulen** [comment]
I cannot ignore `acceptNavigationRequest` because it may be used to hook on anchor navigations. You said that one should use `LinkDelegationPolicy` but it may not work in code that overrides  acceptNavigationRequest` and may break old code.

I can set some special navigation type if only fragment changed in the URL

**Comment by Emdek** [comment]
@annulen, I'm not sure about special type, that would break my workaround now. :-D
What about documenting that part instead? Suggesting to compare URL without fragments.
Also I think that this case with line numbers on github is special, since it's not click but JS triggered by click (otherwise I would expect `QWebPage::NavigationTypeLinkClicked`, plus AFAIK these are not links anyway).

**Comment by annulen** [comment]
>that would break my workaround now

So no changes to avoid breaking code.

>What about documenting that part instead?

+1

>Also I think that this case with line numbers on github is special, since it's not click but JS triggered by click

```
virtual bool QWebPagePrivate::acceptNavigationRequest(QWebFrameAdapter *, const QNetworkRequest &, int)  QUrl( "https://github.com/annulen/webkit/blob/qtwebkit-stable/Makefile#L17" )  5
main frame - didCommitLoadForFrame
main frame - didFinishLoadForFrame
main frame - didChangeLocationWithinPageForFrame
main frame - dispatchDidReplaceStateWithinPage
main frame - didCommitLoadForFrame
main frame - didFinishLoadForFrame
```
Looks similar to what normal anchor click does + history.replaceState()

**Comment by Emdek** [comment]
> So no changes to avoid breaking code.

Except of killing these `loadFinished()` signals, right?

**Comment by annulen** [comment]
Right, relying on stray `loadFinished` doesn't seem as a legitimate use case

**Comment by Emdek** [comment]
OK, it's fine with me.
